signer/src/storage/model.rs

```diff
@@ -93,12 +93,22 @@ impl From<&crate::message::SweepTransactionInfo> for SweepTransaction {
 
 impl utxo::GetFees for Vec<SweepTransaction> {
     /// Return the total fee of all the transactions in the vector.
-    fn get_fees(&self) -> Fees {
-        let total = self.iter().map(|tx| tx.fee).sum();
-        let total_size: u32 = self.iter().map(|tx| tx.vsize).sum();
+    fn get_fees(&self) -> Result<Fees, Error> {
+        let total: u64 = self
+            .iter()
+            .map(|tx| tx.fee)
+            .try_fold(0u64, |acc, fee| acc.checked_add(fee))
+            .ok_or(Error::ArithmeticOverflow)?;
+
+        let total_size: u64 = self.iter().map(|tx| tx.vsize as u64).sum();
+
+        if total_size == 0 {
+            return Err(Error::DivideByZero);
+        }
```

Comment:
This can only happen if the vector is empty, so we could just return zero.

Reply:
I don't think zero is what we want -- `compute_transaction_fee()` matches on an `Option<>` and `None` would not be the same as `Fees::ZERO`. 

But I changed this to return `None` ([891e46a](https://github.com/stacks-network/sbtc/pull/810/commits/891e46aa7ce8aaf05a0fb7e87912aca2a5e96b05)) if `total_size` is zero since I could describe that mode in the trait description and it matches its use. Yeah it should only happen if the vec is empty in prod code, but it's still not panic-safe by simply checking `self.len()` (you could create items all of which have `vsize == 0`).

Reply:
Great points!